### PR TITLE
Auth bypass ids for editions

### PR DESCRIPTION
**Pull request by baisa**
Ensure all newly created Whitehall editions are stored with auth bypass ids.

[Trello card](https://trello.com/c/MBi8s6Lh/374-ensure-all-newly-created-whitehall-editions-are-stored-with-auth-bypass-ids)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance]
(https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -10,50 +10,49 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema[6.1].define(version: 2021_10_19_095600) do
```
Comment (brucebolt) [inline db/schema.rb:13]: There seem to be quite a few unrelated changes to the `schema.rb` file that appear related to the Rails 7 upgrade.  Maybe we missed a step during the upgrades?  In any case, this probably ought to be in another commit (or ideally PR) to keep those changes separate.

File: app/presenters/publishing_api/base_item_presenter.rb [hunk 2]
```diff
@@ -18,6 +18,7 @@ def base_attributes
         publishing_app: "whitehall",
         redirects: [],
         update_type: update_type,
+        auth_bypass_ids: [item.auth_bypass_id]
```
Comment (brucebolt) [inline app/presenters/publishing_api/base_item_presenter.rb:21]:
```suggestion
        auth_bypass_ids: [item.auth_bypass_id],
```
You'll need a comma here to appease rubocop.

**Review by brucebolt — COMMENTED** [review]
Changes to add the `auth_bypass_id` look good to me 👍 , although there's some Rails 7 stuff that ideally should be another PR or commit.

File: test/factories/editions.rb [hunk 3]
```diff
@@ -8,6 +8,7 @@
     change_note { "change-note" }
     summary { "edition-summary" }
     previously_published { false }
+    auth_bypass_id { "52db85fc-0f30-42a6-afdd-c2b31ecc6a67" }
```
Comment (brucebolt) [inline test/factories/editions.rb:11]: Should this also be a random value?

File: db/schema.rb (hunk 1 above)
Comment (baisa) [reply db/schema.rb]: Good point, thank you @brucebolt

File: app/models/edition.rb [hunk 4]
```diff
@@ -643,6 +644,10 @@ def set_public_timestamp
                             end
   end
 
+  def set_auth_bypass_id
+    self.auth_bypass_id = SecureRandom.uuid
```
Comment (kevindew) [inline app/models/edition.rb:648]: Just a quick comment as I know you're still working on this, we shouldn't have this set on before_save as it'll mean every edit to the document changes the auth_bypass_id and thus invalidates the token.

Comment (kevindew) [inline app/models/edition.rb:648]: Just a quick comment as I know you're still working on this, we shouldn't have this set on before_save as it'll mean every edit to the document changes the auth_bypass_id and thus invalidates the token.

File: app/presenters/publishing_api/case_study_presenter.rb [hunk 5]
```diff
@@ -24,6 +24,7 @@ def content
           public_updated_at: item.public_timestamp || item.updated_at,
           rendering_app: Whitehall::RenderingApp::GOVERNMENT_FRONTEND,
           schema_name: "case_study",
+          auth_bypass_ids: [item.auth_bypass_id],
```
Comment (kevindew) [inline app/presenters/publishing_api/case_study_presenter.rb:27]: @baisa, just a thought based off our chat last week. Code changes like this will cause problems until we've backfilled all the past editions with auth_bypass_ids. 

You might need to break this up into a few PRs:

1) create auth_bypass_ids on new editions
2) backfill old editions with auth_bypass_ids
3) set auth_bypass_ids to non null
4) send them to Publishing API

Comment (BeckaL) [reply app/presenters/publishing_api/case_study_presenter.rb:27]: Hey Kevin, I can't see anywhere in publishing api that suggests it doesn't accept arrays with nil, and e.g. this [test data](https://github.com/alphagov/publishing-api/blob/dd8dd9232d3cbf33b8945fdd898ebe80d7dcfcf6/spec/support/request_helpers/mocks.rb#L53) suggests that it can. (Also just wrote a Command test to see, but don't know what it's mocking out).

Am I missing something here? (Imagine this is v probable :-) ) Original thinking was that it'd be nice to be able to deploy a fully end to end thing, but more than happy to split it out (and obviously need to do it if it's not safe).
